### PR TITLE
JENKINS-28629 - Fix Apache Maven 3.3 support - introduce maven 3.3 an…

### PR DESCRIPTION
**Pull request by tomerc**
…d pass toolchains capability to it

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,8 +44,8 @@ THE SOFTWARE.
 
   <properties>
     <jenkins.version>1.580.1</jenkins.version>
-    <java.level>6</java.level>
-    <mavenInterceptorsVersion>1.7</mavenInterceptorsVersion>  
+    <java.level>7</java.level>
```
Comment (aheritier) [inline pom.xml]: I know why you are doing this for the maven compatibility for 3.3 which requires j7 but in that case we should update the minimal jenkins version to Jenkins OSS >= 1.612 or Jenkins LTS >= 1.625.1
We should avoid older versions of jenkins which may run on J6 use this version
In that case I think that we should release a new version of the plugin before integrating this change

Comment (tomerc) [inline pom.xml]: Ok, would you like me to update the jenkins version as part of this PR?

Comment (aheritier) [inline pom.xml]: let's wait for others feedbacks .... like @olamy, @jglick ...

Comment (maximusfloydus) [inline pom.xml]: Any progress on this change?

Comment (aheritier) [inline pom.xml]: @maximusfloydus I just merged https://github.com/jenkinsci/maven-interceptors/pull/9 I need to apply few more changes on it to release it (this week). After this I'll be able to use it here and to merge this PR

**Comment by nfalco79** [comment]
any chance to release this?

**Comment by nfalco79** [comment]
This patch works passing custom settings.xml defined in config-file-provider plugin but use toolschain.xml in .m2 folder instead that one provided in config-file-provider.

**Comment by aheritier** [comment]
thanks for this feedback @nfalco79 I will investigate

**Comment by nfalco79** [comment]
Jenkins 1.625.3
maven-plugin PR 63
job of type Maven

following a partial log

```
[bundles] $ /home/rad/.jenkins/tools/hudson.model.JDK/JDK_1.8/bin/java -cp /home/rad/.jenkins/plugins/maven-plugin/WEB-INF/lib/maven33-agent-1.8-SNAPSHOT.jar:/home/rad/.jenkins/tools/hudson.tasks.Maven_MavenInstallation/Maven_3.3/boot/plexus-classworlds-2.5.2.jar:/home/rad/.jenkins/tools/hudson.tasks.Maven_MavenInstallation/Maven_3.3/conf/logging jenkins.maven3.agent.Maven33Main /home/rad/.jenkins/tools/hudson.tasks.Maven_MavenInstallation/Maven_3.3 /home/rad/.jenkins/war/WEB-INF/lib/remoting-2.53.2.jar /home/rad/.jenkins/plugins/maven-plugin/WEB-INF/lib/maven33-interceptor-1.8-SNAPSHOT.jar /home/rad/.jenkins/plugins/maven-plugin/WEB-INF/lib/maven3-interceptor-commons-1.8-SNAPSHOT.jar 49785
<===[JENKINS REMOTING CAPACITY]===>channel started
using settings config with name MySettings
Replacing all maven server entries not found in credentials list is true
Executing Maven:  -B -f /home/rad/.jenkins/jobs/new-build-system/workspace/bundles/pom.xml -s /tmp/settings6449525320507493383.xml clean install
[INFO] Scanning for projects...
```

**Comment by aheritier** [comment]
I published SNAPSHOTs of maven-interceptors to try to validate the build on CI env (locally I don't succeed to pass all tests for now)

**Comment by aheritier** [comment]
Tests are passing thus I propose to merge the PR and to iterate to fix existing bugs before the release.

https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/237/

WDYT ? (FYI @olamy is working on the test framework which should ease to manage them)

If you are interested by this fix you can give a try to this SNAPSHOT :
https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/237/org.jenkins-ci.main$maven-plugin/artifact/org.jenkins-ci.main/maven-plugin/2.14-SNAPSHOT/maven-plugin-2.14-SNAPSHOT.hpi

**Comment by tomerc** [comment]
I like the idea of iterating on it. +1 from me.

The 2.14 snapshot so far looks good.

**Pull request merged**